### ISSUE TITLE
OCRmyPDF appends a space to each text element at the end of the line

### ISSUE THREAD

**Issue by gowallasnewpony** [ocrmypdf/OCRmyPDF#1131]
### Describe the bug

I get blank spaces at the end of all the last words of a line.

Is there a way to not have blank spaces at the end of all the last words of a line?

Natively born PDF don't do this ...

The problem is that my bank which is scanning documents will not recognize text correctly. The problem is only the space after the last word of each line.

Thank you for your great software!

### Steps to reproduce

```plain text
1. ocrmypdf --language deu "$input_file" "$output_file"
```

### OCRmyPDF version

14.1.0

**Comment by jbarlow83** [comment]
What version of Tesseract?

**Comment by gowallasnewpony** [comment]
Thank you jbarlow83

tesseract 5.3.0
 leptonica-1.83.0
  libgif 5.2.1 : libjpeg 6b (libjpeg-turbo 2.1.5.1) : libpng 1.6.39 : libtiff 4.5.1 : zlib 1.2.13 : libwebp 1.3.1 : libopenjp2 2.5.0
 Found AVX2
 Found AVX
 Found FMA
 Found SSE4.1
 Found OpenMP 201511

**Comment by jbarlow83** [comment]
The extra spaces come from Tesseract.

In a quick test of modifying this behavior in Tesseract's PDF renderers, Evince PDF viewer's highlighting of text was less accurate without the trailing space. Having worked on Tesseract as well to improve rendering, it's a hard sell to get any change through that's not an improve the across the board in appearance and extractable output text.

Standard Tesseract
![image](https://github.com/ocrmypdf/OCRmyPDF/assets/1825843/00168166-81e7-4189-a829-bcd3237cfc23)

Modified version - EOL spaces suppressed
![image](https://github.com/ocrmypdf/OCRmyPDF/assets/1825843/497b670e-9b22-442d-a4c0-1efc7a886252)

Next step - try suppressing the Unicode character but still increasing the word width.

**Comment by gowallasnewpony** [comment]
Thank you jbarlow83, for this comparison!

Ok, I can see the problem ... 

What is to change with tesseract when it's more important not having EOL spaces?

In my situation not having EOL spaces would allow my bank to scan documents and import their content without problems.

**Comment by jbarlow83** [comment]
My experiment was the following patch to Tesseract's main branch at commit [ce9369237022a3a16b880e2bd4ae2f4639284469
](https://github.com/tesseract-ocr/tesseract/commit/ce9369237022a3a16b880e2bd4ae2f4639284469)
if you want to play with it.

```diff
diff --git a/src/api/pdfrenderer.cpp b/src/api/pdfrenderer.cpp
index 81cf2e24..e251a4af 100644
--- a/src/api/pdfrenderer.cpp
+++ b/src/api/pdfrenderer.cpp
@@ -464,12 +478,13 @@ char *TessPDFRenderer::GetPDFTextObjects(TessBaseAPI *api, double width, double
       }
       res_it->Next(RIL_SYMBOL);
     } while (!res_it->Empty(RIL_BLOCK) && !res_it->IsAtBeginningOf(RIL_WORD));
-    if (res_it->IsAtBeginningOf(RIL_WORD)) {
+    if (res_it->IsAtBeginningOf(RIL_WORD) && !last_word_in_line) {
       pdf_word += "0020";
       pdf_word_len++;
     }
     if (word_length > 0 && pdf_word_len > 0) {
```

**Comment by jbarlow83** [comment]
Will be fixed in #1194
